tensorflow/lite/micro/kernels/squared_difference.cc

```diff
@@ -124,6 +124,56 @@ TfLiteStatus SquaredDifferencePrepare(TfLiteContext* context,
         std::numeric_limits<int8_t>::min();
     data->arithmetic_params.quantized_activation_max =
         std::numeric_limits<int8_t>::max();
+  } else if (input1->type == kTfLiteInt16) {
```

Comment:
only the TF_LITE_ENSUREs are different between int8 and int16, right? And the last QuantizeMultipler.

Can we share the rest of the code? We will still need an if for quantized type vs float, right?

If it helps you can break up the logic into a PrepareQuantized that only gets called if input type is not a float.

Reply:
Done.